src/rust/plugin-registry/src/server/s3_url.rs

```diff
@@ -1,13 +1,11 @@
 use grapl_config::env_helpers::ENV_ENDPOINT;
 
-/// Some discussion about this function here:
-/// https://grapl-internal.slack.com/archives/C02J5JYS92S/p1648667074625629
 pub fn get_s3_url(bucket: &str, key: &str) -> String {
     let local_endpoint = std::env::var(ENV_ENDPOINT).ok();
     // If the above is specified, we're running locally against Localhost
     // Otherwise, it's against prod S3; so we can use virtual-hosted-style URLs
     local_endpoint.map_or_else(
-        || format!("http://{bucket}.s3.amazonaws.com/{key}"),
+        || format!("s3::{bucket}.s3.amazonaws.com/{key}"),
```

Comment:
How is this a URL? I would buy `s3://...` but this just looks _weird_.

Reply:
Ah let me change the function name to URI. And agreed that this is a very weird schema. It feels like someone was like, well our arns already use `::` as separators, why not just reuse that?

Reply:
Out of curiosity, would it have worked with `https://` instead of `http://`?

Reply:
Sadly no. I tried that early on 😿

Reply:
So apparently `s3://` does work and is officially supported by AWS. I've switched to that, especially since this is the format the s3 URI link in the console is (https://s3.console.aws.amazon.com/s3/object/plugin-registry-bucket-b1abfd2?region=us-east-1&prefix=plugins/tenant_id_024885b9-dcf0-469d-b12a-81876cc045c8/plugin_type-analyzer/51f8b573-6b7c-411c-9379-2b1070205c1f.bin and search s3 URI).
Apparently, the `s3::` was something the go-getter lib created and call [Forced Protocols](https://github.com/hashicorp/go-getter#forced-protocol). Basically its a way of saying that this is a s3 url.